pkg/test/framework/components/authz/provider.go

```diff
@@ -0,0 +1,205 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authz
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"istio.io/istio/pkg/config/protocol"
+	echoClient "istio.io/istio/pkg/test/echo"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/check"
+	"istio.io/istio/pkg/test/framework/components/echo/match"
+)
+
+// API used by a Provider. Either HTTP or GRPC.
+type API string
+
+const (
+	HTTP API = "http"
+	GRPC API = "grpc"
+)
+
+// Provider for authz requests.
+type Provider interface {
+	Name() string
+
+	// API used by this provider.
+	API() API
+
+	// IsProtocolSupported returns true if the given request protocol is supported by this provider.
+	IsProtocolSupported(protocol.Instance) bool
```

Comment:
Are these methods actually used in the test? I found only the `Name()` and `API()` is used. If they are needed in a follow-up PR, it's better to add in the follow-up PR together for easier review.